### PR TITLE
Extra xcodebuild arguments

### PR DESCRIPTION
**Pull request by atanas-bitrise**
### Checklist

- [x] I've read and accepted the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- Requires MINOR [version update](https://semver.org/)

### Context

No way to pass additional arguments to the XCode Analyze step

Resolves https://bitrise.atlassian.net/browse/STEP-1192

### Changes

Reused code from the Xcode archive step.

### Investigation details

No alternatives were considered.

### Decisions

Use the code from the archive step by analogy and adapt it were needed.

### CODE REVIEW

File: step.yml [hunk 1]
```diff
@@ -151,6 +151,14 @@ inputs:
...
       - "swift_packages"
       is_required: true
+  - xcodebuild_options:
+    opts:
+      category: Debug
+      title: Additional options for xcodebuild call
+      description: |-
+        Options added to the end of the xcodebuild call.
+        You can use multiple options, separated by a space
+        character. Example: `-xcconfig PATH -verbose`
```
Comment (DamienBitrise) [inline step.yml:161]: Should we add some info about how to make the .xcactivitylog file get generated maybe?

`"-derivedDataPath" "$BITRISE_SOURCE_DIR/build/DerivedData" -resultBundlePath "$BITRISE_SOURCE_DIR/build/ResultBundle" OTHER_SWIFT_FLAGS="-Xfrontend -debug-time-function-bodies"`

Would it even make sense to set these as the default additional args?

Comment (mateherber) [reply step.yml:161]: This input is more generic than "generating the xcactivitylog" so its description should be generic as well. Maybe as an "example" we can provide that scenario tho

Comment (atanas-bitrise) [reply step.yml:161]: Maybe I can replace the example `-xcconfig PATH -verbose` with `-derivedDataPath DERIVED_DATA_PATH -resultBundlePath RESULT_BUNDLE_PATH OTHER_SWIFT_FLAGS="-Xfrontend -debug-time-function-bodies"`?

Comment (lpusok) [reply step.yml:159]: @DamienBitrise @atanas-bitrise the resultBundlePath is enough to trigger the generation of the xcactivity file.
Decided to add the `-resultBundlePath` in all cases in the Step code.

**Review by Bence1001 — APPROVED** [review]
LGTM!

**Pull request merged**